core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -493,29 +510,38 @@ private void updateMetrics(boolean success, Instant startTime,
    * @param path The path to sync
    * @return true if this path was synced
    */
-  private boolean processSyncPath(AlluxioURI path) {
+  private SyncResult processSyncPath(AlluxioURI path) {
     if (path == null) {
-      return false;
+      return SyncResult.INVALID_RESULT;
     }
+
+    // if we have already loaded the path from the UFS, and the path
+    // is not a directory, then we will always finish the sync
+    // (even if it is not needed) since we already have all the data we need
+    boolean forceSync = mStatusCache.hasStatus(path).map(
```

Comment:
This isn't quite true, as with UFS that have ACL enabled we will still need an extra UFS operation to get the ACL. So maybe this option should only be enabled for UFS that only have ACL disabled?

Reply:
Actually I updated this, so that if ACL is enabled and a sync is not needed on a child file then the sync is skipped. Only if ACL is disabled will the sync still happed if it is not needed, because this will not need an extra UFS operation.